src/catalog/database_catalog.cpp

```diff
@@ -406,39 +400,41 @@ bool DatabaseCatalog::CreateColumn(transaction::TransactionContext *const txn, c
   bool UNUSED_ATTRIBUTE result = columns_oid_index_->InsertUnique(txn, *pr, tupleslot);
   TERRIER_ASSERT(result, "Assigned OIDs failed to be unique.");
 
-  // Step 4: Insert into class index
-  const auto class_pri = columns_class_index_->GetProjectedRowInitializer();
-  pr = class_pri.InitializeRow(buffer);
-  // Write the attributes in the ProjectedRow
-  *(reinterpret_cast<ClassOid *>(pr->AccessForceNotNull(0))) = class_oid;
-
-  result = columns_class_index_->Insert(txn, *pr, tupleslot);
-  TERRIER_ASSERT(result, "Assigned OIDs failed to be unique.");
-
   // Finish
   delete[] buffer;
   return true;
 }
 
 template <typename Column, typename ClassOid, typename ColOid>
-std::vector<Column> DatabaseCatalog::GetColumns(transaction::TransactionContext *const txn, const ClassOid class_oid) {
+std::vector<Column> DatabaseCatalog::GetColumns(transaction::TransactionContext *const txn, ClassOid class_oid) {
   // Step 1: Read Index
   const std::vector<col_oid_t> table_oids{ATTNUM_COL_OID, ATTNAME_COL_OID,    ATTTYPID_COL_OID,
                                           ATTLEN_COL_OID, ATTNOTNULL_COL_OID, ADSRC_COL_OID};
   // NOLINTNEXTLINE
   auto [table_pri, table_pm] = columns_->InitializerForProjectedRow(table_oids);
-  const auto class_pri = columns_class_index_->GetProjectedRowInitializer();
+  const auto oid_pri = columns_oid_index_->GetProjectedRowInitializer();
   // Buffer is large enough to hold all prs
   byte *const buffer = common::AllocationUtil::AllocateAligned(table_pri.ProjectedRowSize());
+  byte *const key_buffer = common::AllocationUtil::AllocateAligned(oid_pri.ProjectedRowSize());
   // Scan the class index
-  auto *pr = class_pri.InitializeRow(buffer);
+  auto *pr = oid_pri.InitializeRow(buffer);
+  auto *pr_high = oid_pri.InitializeRow(key_buffer);
+
   // Write the attributes in the ProjectedRow
+  // Low key (class, INVALID_COLUMN_OID) [using uint32_t to avoid adding ColOid to template]
   *(reinterpret_cast<ClassOid *>(pr->AccessForceNotNull(0))) = class_oid;
+  *(reinterpret_cast<uint32_t *>(pr->AccessForceNotNull(1))) = 0;
```

Comment:
`oid_pm` is used to insert into the index (in CreateColumn), but not to read from it (in GetColumns & DeleteColumns). These two uses should be the consistent. Probably don't need the map in CreateColumn?